### PR TITLE
bgpv2: fix pod ip pool cleanup

### PR DESCRIPTION
**Pull request by harsimran-pabla**
Modifying the reconciler logic in pod IP pool reconciler. Metadata to be maintained per pool, which is used to remove paths for pools which are deleted. Fixed the tests to match new metadata data structure.

### CODE REVIEW

File: pkg/bgpv1/manager/reconcilerv2/pod_ip_pool.go [hunk 1]
```diff
@@ -85,25 +89,94 @@ func (r *PodIPPoolReconciler) Reconcile(ctx context.Context, p ReconcileParams)
...
+}
+
+func (r *PodIPPoolReconciler) reconcilePaths(ctx context.Context, p ReconcileParams, desiredPeerAdverts PeerAdvertisements, lp map[string][]netip.Prefix) error {
+	poolsAFPaths, err := r.getDesiredPoolAFPaths(p, desiredPeerAdverts, lp)
+	if err != nil {
+		return err
+	}
+
+	for poolKey, desiredPoolAFPaths := range poolsAFPaths {
+		metadata := r.getMetadata(p.BGPInstance)
```
Comment (rastislavs) [inline pkg/bgpv1/manager/reconcilerv2/pod_ip_pool.go]: nit: this can be moved out of the range loop

**Review by rastislavs — APPROVED** [review]
LGTM, just a small nit

**Review by ldelossa — APPROVED** [review]
LGTM

**Pull request merged**